### PR TITLE
Model changes

### PR DESCRIPTION
**Pull request by subutai**
Added profiling using Network API's built in profiler.
Temporary tokenization to avoid caching
Simple label classification script
Minor changes

### CODE REVIEW

**Comment by subutai** [comment]
@BoltzmannBrain

File: projects/nlp/simple_labels.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+helpStr = """
+Simple script to run a labeled dataset.
+
+Example invocations:
+
+python hello_classification_model.py -m keywords --dataPath FILE --numLabels 3
```
Comment (BoltzmannBrain) [inline projects/nlp/simple_labels.py]: simple_labels.py, not hello_classification_model.py

File: htmresearch/support/csv_helper.py [hunk 2]
```diff
@@ -22,6 +22,8 @@
 This file contains CSV utility functions to use with nupic.fluent experiments.
 """
 
+import itertools
```
Comment (BoltzmannBrain) [inline htmresearch/support/csv_helper.py]: alpha order please

File: htmresearch/frameworks/nlp/classify_document_fingerprint.py [hunk 3]
```diff
@@ -264,6 +286,30 @@ def printRegionOutputs(self):
...
+    profileInfo = []
+    for region in self.network.regions.values():
+      timer = region.computeTimer
+      profileInfo.append([region.name,
+                          timer.getStartCount(),
+                          timer.getElapsed(),
+                          100.0*timer.getElapsed()/totalTime])
+
+    profileInfo.append(["Total time", "",totalTime,"100.0"])
+    print tabulate(profileInfo, headers=["Region", "Count",
```
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/classify_document_fingerprint.py]: indenting here is strange (also in htmresearch/frameworks/nlp/classify_htm.py)

File: htmresearch/frameworks/nlp/classify_document_fingerprint.py [hunk 4]
```diff
@@ -264,6 +286,30 @@ def printRegionOutputs(self):
...
+
+    profileInfo = []
+    for region in self.network.regions.values():
+      timer = region.computeTimer
+      profileInfo.append([region.name,
+                          timer.getStartCount(),
+                          timer.getElapsed(),
+                          100.0*timer.getElapsed()/totalTime])
+
+    profileInfo.append(["Total time", "",totalTime,"100.0"])
```
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/classify_document_fingerprint.py]: spaces after commas (also in htmresearch/frameworks/nlp/classify_htm.py)

File: htmresearch/frameworks/nlp/classify_document_fingerprint.py [hunk 5]
```diff
@@ -142,6 +145,25 @@ def _initializeRegionHelpers(self):
...
+    method.
+
+    @param text         (str)     A bunch of text.
+    @param preprocess   (bool)    Whether or not to preprocess the text data.
+    """
+    encoder = self.sensorRegion.getSelf().encoder
+    sentenceList = encoder.client.tokenize(text)
+    tokenList = []
+    for sentence in sentenceList:
+      tokenList.extend(sentence.split(','))
```
Comment (BoltzmannBrain) [inline htmresearch/frameworks/nlp/classify_document_fingerprint.py]: double quotes please (also in htmresearch/frameworks/nlp/classify_htm.py)

**Comment by BoltzmannBrain** [comment]
@subutai HTM models don't run b/c of a bug in the recent cache path changes: #396. Given they do run after the fix, :+1: (besides a few nitpicks).

**Pull request merged**